adapters/src/main/kotlin/com/krzykrucz/elesson/currentlesson/preparedness/adapters/readmodel/PreparednessInSemesterReadModel.kt

```diff
@@ -0,0 +1,61 @@
+package com.krzykrucz.elesson.currentlesson.preparedness.adapters.readmodel
+
+import arrow.core.Option
+import arrow.core.getOrElse
+import arrow.effects.IO
+import com.krzykrucz.elesson.currentlesson.preparedness.adapters.readmodel.StudentInSemesterReadModel.READ_MODEL
+import com.krzykrucz.elesson.currentlesson.preparedness.domain.StudentMarkedUnprepared
+import com.krzykrucz.elesson.currentlesson.preparedness.readmodel.GetStudentSubjectUnpreparednessInASemester
+import com.krzykrucz.elesson.currentlesson.preparedness.readmodel.StudentInSemester
+import com.krzykrucz.elesson.currentlesson.preparedness.readmodel.StudentSubjectUnpreparednessInASemester
+import com.krzykrucz.elesson.currentlesson.preparedness.readmodel.WriteUnpreparednessInTheRegister
+import com.krzykrucz.elesson.currentlesson.shared.AsyncFactory
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.event.EventListener
+import java.util.concurrent.ConcurrentHashMap
+
+class ReadModelProjector : WriteUnpreparednessInTheRegister {
```

Comment:
as it uses events to save read model it's not `Projector` as it doesn't use original source but saves some transformed version upon on events